### PR TITLE
sdk metrics sample

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

Adds sdk metrics sample which is similar to the Go sample: https://github.com/temporalio/samples-go/tree/main/metrics

### CODE REVIEW

File: src/main/java/io/temporal/samples/metrics/MetricsStarter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            .setWorkflowId("metricsWorkflow")
+            .setTaskQueue(DEFAULT_TASK_QUEUE_NAME)
+            .build();
+    MetricsWorkflow workflow = client.newWorkflowStub(MetricsWorkflow.class, workflowOptions);
+
+    String result = workflow.exec("hello metrics");
+
+    System.out.println("Result: " + result);
+
+    System.out.println("Check metrics at http://localhost:8080/sdkmetrics");
```
Comment (mfateev) [inline src/main/java/io/temporal/samples/metrics/MetricsStarter.java:41]: I thought that starter also should report metrics

File: src/main/java/io/temporal/samples/metrics/MetricsWorker.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    // Stopping the worker will stop the http server that exposes the
+    // scrape endpoint.
+  }
+
+  /**
+   * Starts HttpServer to expose a scrape endpoint. See
+   * https://micrometer.io/docs/registry/prometheus for more info.
+   */
+  private static void startServer() {
```
Comment (mfateev) [inline src/main/java/io/temporal/samples/metrics/MetricsWorker.java:76]: I would rename it to something more descriptive like `startPrometheusScrapeEndpoint`

Comment (tsurdilo) [reply src/main/java/io/temporal/samples/metrics/MetricsWorker.java]: renamed.

**Comment by tsurdilo** [comment]
@mfateev updated

**Review by Spikhalskiy — APPROVED** [review]

**Pull request merged**